modules/monitoring-exposing-custom-application-metrics-for-horizontal-pod-autoscaling.adoc

```diff
@@ -0,0 +1,223 @@
+// Module included in the following assemblies:
+//
+// monitoring/monitoring.adoc
+
+[id='exposing-custom-application-metrics-for-horizontal-pod-autoscaling-{context}']
+= Exposing custom application metrics for horizontal pod autoscaling
+
+This procedure shows, on an example, how a human operator for the cluster can, using `prometheus-adapter`, expose custom application metrics for the horizontal pod autoscaler.
+
+.Prerequisites
+
+* Make sure you have configured cluster for application monitoring. In this example, it is presumed that Prometheus and Alertmanager instances were installed in the `default` namespace.
+* Make sure you have configured monitoring for your application application monitoring. In this example, it is presumed that the application and the service monitor for it were installed in the `default` namespace.
+
+.Procedure
+
+The following steps present snippets of YAML files. For each step, put the snippet into a YAML file, for example, `deploy.yaml` and run this command to apply the configuration file to the cluster:
+
+  $ oc apply -f deploy.yaml
+
+. Create the ??? configuration for `prometheus-adapter`:
```

Comment:
These are several different RBAC-related resources in this block. Split them out.